packages/react-dom/src/__tests__/CSSPropertyOperations-test.js

```diff
@@ -186,6 +186,36 @@ describe('CSSPropertyOperations', () => {
     ]);
   });
 
+  it('should warn about style containing !important', () => {
+    class Comp extends React.Component {
+      static displayName = 'Comp';
+
+      render() {
+        return (
+          <div
+            style={{
+              backgroundColor: 'red!important',
+              color: 'green!important',
+            }}
+          />
+        );
+      }
+    }
+
+    const root = document.createElement('div');
+
+    expect(() => ReactDOM.render(<Comp />, root)).toWarnDev([
+      'Warning: Style property values shouldn\'t contain "!important". ' +
+        'Try "backgroundColor: red" instead.' +
+        '\n    in div (at **)' +
+        '\n    in Comp (at **)',
```

Comment:
I would prefer a wording like

```
Warning: React does not support `!important` in inline style values. The specified `backgroundColor: "red!important"` style will be ignored. Either remove `!important` from the style value, or use a DOM ref to directly to set this style value.
```

Reply:
Are you asking for that exact wording? Do you mind if we clear up the ambiguities? React / ReactDOM, style values / style property values